packages/glimmer-runtime/lib/environment.ts

```diff
@@ -116,9 +116,15 @@ export class Scope {
   }
 }
 
+interface ModifierPair {
+  manager: ModifierManager<Opaque>;
+  element: Element;
+}
+
 export abstract class Environment {
   protected updateOperations: DOMChanges;
   protected appendOperations: DOMTreeConstruction;
+  private createdModifierManagers: ModifierPair[] = [];
```

Comment:
It appears we need to set this to `[]` instead of `null` do to where modifiers are created during evaluation.